estate/models/estate_property.py

```diff
@@ -0,0 +1,111 @@
+        ('offer_received','OFFER RECEIVED'),
+        ('offer_accepted','OFFER ACCEPTED'),
+        ('sold','SOLD'),
+        ('cancelled','CANCELLED'),
+        ], default='new')
+    property_type_id = fields.Many2one('estate.property.type', string='Real Estate Type')
+    buyer_id = fields.Many2one('res.partner', string='Buyer',)
+    salesperson_id = fields.Many2one('res.partner', string='Salesperson', default=lambda self: self.env.user)
+    tag_ids = fields.Many2many('estate.property.tag', string='Real Estate Tag')
+    offer_ids = fields.One2many('estate.property.offer', inverse_name='property_id')
+    total_area = fields.Float('Total Area (sqm)', compute='_compute_total_area', readonly=True)
+    best_offer = fields.Float('Best Offer', compute='_compute_best_offer', readonly=True)
+    type_id = fields.Many2one('estate.property.type')
+    _order = 'id desc'
+
+    _sql_constraints = [
+        ('expected_price', 'CHECK(expected_price > 0)',
+         'The expected price should be strictly grater than 0.'),
+        ('selling_price', 'CHECK(selling_price > 0)',
+         'The selling price should be strictly grater than 0.'),
+        ]
+
+
+    def action_open_offers(self):
+        self.ensure_one()
+        return {
+            'name': 'Property Offer',
+            'views': [(self.env.ref('estate.estate_property_offer_view_list').id, 'list')],
+            'type': 'ir.actions.act_window',
+            'domain' : [('id', 'in', self.offer_ids.ids)],
+            'res_model': 'estate.property.offer',
+        }
+
+    @api.ondelete(at_uninstall=False)
+    def on_delete(self, vals_list):
+        for val in vals_list:
+            if val and val['state'] == 'new' or val['state'] == 'cancelled':
+                raise UserError('You can not delete a new or cancelled property')
+        
+
+    @api.constrains('expected_price','selling_price')
+    def _check_expected_price(self):
+        for record in self:
+            if float_compare(record.expected_price * 0.9, record.selling_price, 3) == 1 and record.selling_price != 0:
+                raise ValidationError(f"The selling price must be at least the 90% of the expected price.")
+
+
+    def action_cancel(self):
+        if self.state != 'cancelled':
+            self.state = 'cancelled'
+
+    def action_sold(self):
+        if self.state == 'cencelled':
+            raise UserError("You can't sold an estate marked as CANCELLED")
+        self.state = 'sold'        
+
+    
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + (record.garden_area or 0)
+
+    @api.depends("offer_ids.price")
+    def _compute_best_offer(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_offer = max(offer.price for offer in record.offer_ids)
+            else:
+                record.best_offer = 0  # Set to 0 if no offers are present
```

Comment:
It's good to want to add comments for better readability for the one that might look or review your code, but they should be reserved for when the use case is hard to comprehend. You can leave these ones, but your code is well explicit enough so you don't need to bother next time :smile: (but please write some is the code gets more complicated)
Just to complete what I write here: comments should explain *why* the code is needed, in complicated cases, not *what* it does, the code should be self-explanatory for the last part